airbyte-local.sh

```diff
@@ -134,22 +79,37 @@ function writeSrcConfig() {
           ({}; . + {($a[2*$i]): ($a[2*$i + 1]|fromjson? // .)})' > "$tempdir/$src_config_filename"
 }
 
-# TODO: take catalog as input parameter
 function writeSrcCatalog() {
-    case $src_docker_image in
-        *pagerduty*)
-            echo $pagerduty_src_catalog | jq > $tempdir/$src_catalog_filename
-            stream_prefix=mypagerdutysrc__pagerduty__
-            ;;
-        *servicenow*)
-            echo $servicenow_src_catalog | jq > $tempdir/$src_catalog_filename
-            stream_prefix=myservicenowsrc__servicenow__
-            ;;
-        *)
-            echo "Error: $src_docker_image is currently not supported"
-            exit 1
-            ;;
-    esac
+    supported_src="pagerduty|servicenow|phabricator"
+    src_docker_image_regex="^.*airbyte-($supported_src)-source$"
+    if [[ "$src_docker_image" =~ $src_docker_image_regex ]]; then
+        source_type="$(sed 's/.*airbyte-\(.*\)-source/\1/' <<< "$src_docker_image")"
```

Comment:
our convention allows us to extract the source_type and use it to retrieve the necessary catalog and construct the stream prefix.